### ISSUE TITLE
getWindowIndex() sometime is delay

### ISSUE THREAD

**Issue by RuijiePan** [google/ExoPlayer#4547]
//Version
2.7.3

//Description
I'm using ClippingMediaSource and DynamicConcatenatingMediaSource to play multiple video clips, and i need to know the current index of window when onVideoSizeChanged callback, but i found sometime the index is not correct because onPositionDiscontinuity->DISCONTINUITY_REASON_PERIOD_TRANSITION is later than  onVideoSizeChanged.
so i getWindowIndex() is an un correct index.

My question is how to get exact window index when onVideoSizeChanged?

**Comment by tonihei** [comment]
What you describe is basically working as intended. The reason is that these two event (onVideoSizeChanged and onPositionDiscontinuity) belong to different listeners and our listeners are not synchronized to each other for various reasons.

However, we recently added the `AnalyticsListener` class which combines all listeners we have in one place and gives additional information for each event (including which window they belong to). This new listener has been added in 2.8.0. The easiest way to use it is with `SimpleExoPlayer.addAnalyticsListener`.
